CloudFormation loader fails with !GetAtt

## Steps to reproduce

Try to load the following template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: Simple API SAM Template
  
Globals:
  Function:
    Timeout: 3
    MemorySize: 128
    Runtime: python3.12

Resources:
  MyApi:
    Type: AWS::Serverless::Api
    Properties:
      StageName: Prod
      Cors:
        AllowMethods: "'GET,POST,OPTIONS'"
        AllowHeaders: "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token'"
        AllowOrigin: "'*'"

  HelloWorldFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: hello_world/
      Handler: app.lambda_handler
      Events:
        HelloWorld:
          Type: Api
          Properties:
            RestApiId: !Ref MyApi
            Path: /hello
            Method: get

Outputs:
  HelloWorldApi:
    Description: API Gateway endpoint URL for Hello World function
    Value: !Sub "https://${MyApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/hello/"
  HelloWorldFunction:
    Description: Hello World Lambda Function ARN
    Value: !GetAtt HelloWorldFunction.Arn
  HelloWorldFunctionIamRole:
    Description: Implicit IAM Role created for Hello World function
    Value: !GetAtt HelloWorldFunctionRole.Arn
```

## Current behaviour

```
loader = <aws_sam_testing.cfn.CloudFormationLoader object at 0x11421ee40>
node = ScalarNode(tag='!GetAtt', value='HelloWorldFunction.Arn')

    def construct_get_att(loader: yaml.Loader, node: yaml.Node) -> GetAttTag:
        """Construct !GetAtt tag."""
        if not isinstance(node, yaml.SequenceNode):
>           raise yaml.constructor.ConstructorError(
                None,
                None,
                "expected a sequence node, but found %s" % get_node_type_name(node),
                node.start_mark,
            )
E           yaml.constructor.ConstructorError: expected a sequence node, but found scalar
E             in "<unicode string>", line 40, column 12:
E                   Value: !GetAtt HelloWorldFunction.Arn
E                          ^

aws_sam_testing/cfn.py:135: ConstructorError
```

## Expected behaviour

The template is properly loaded with CloudFormation tags

## Root cause

The GetAtt node is not correctly implemented. The `!GetAtt` supports dot notation and array as well as defined at https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/intrinsic-function-reference-getatt.html.

Here's the example:

```
Fn::GetAtt: [ logicalNameOfResource, attributeName ]
```
equals to
```
!GetAtt logicalNameOfResource.attributeName
```

--- a/aws_sam_testing/cfn.py
+++ b/aws_sam_testing/cfn.py
@@ -131,22 +131,37 @@ def construct_ref(loader: yaml.Loader, node: yaml.Node) -> RefTag:
 
 def construct_get_att(loader: yaml.Loader, node: yaml.Node) -> GetAttTag:
     """Construct !GetAtt tag."""
-    if not isinstance(node, yaml.SequenceNode):
+    if isinstance(node, yaml.SequenceNode):
+        # Array notation: !GetAtt [LogicalName, AttributeName]
+        value = loader.construct_sequence(node)
+        if len(value) != 2:
+            raise yaml.constructor.ConstructorError(
+                None,
+                None,
+                "expected 2 items in sequence, but found %d" % len(value),
+                node.start_mark,
+            )
+        return GetAttTag(value)
+    elif isinstance(node, yaml.ScalarNode):
+        # Dot notation: !GetAtt LogicalName.AttributeName
+        value = loader.construct_scalar(node)
+        if not value or "." not in value:
+            raise yaml.constructor.ConstructorError(
+                None,
+                None,
+                "!GetAtt scalar must be in format 'LogicalName.AttributeName'",
+                node.start_mark,
+            )
+        # Split only on the first dot to handle nested attributes
+        parts = value.split(".", 1)
+        return GetAttTag(parts)
+    else:
         raise yaml.constructor.ConstructorError(
             None,
             None,
-            "expected a sequence node, but found %s" % get_node_type_name(node),
+            "expected a sequence or scalar node, but found %s" % get_node_type_name(node),
             node.start_mark,
         )
-    value = loader.construct_sequence(node)
-    if len(value) != 2:
-        raise yaml.constructor.ConstructorError(
-            None,
-            None,
-            "expected 2 items in sequence, but found %d" % len(value),
-            node.start_mark,
-        )
-    return GetAttTag(value)
 
 
 def construct_sub(loader: yaml.Loader, node: yaml.Node) -> SubTag:

--- a/tests/test_cfn.py
+++ b/tests/test_cfn.py
@@ -31,7 +31,7 @@ class TestCloudFormationLoader:
           """,
             {"Resources": {"MyBucket": {"Type": "AWS::S3::Bucket", "Properties": {"BucketName": RefTag("MyBucketName")}}}},
         ),
-        # GetAtt tag tests
+        # GetAtt tag tests - array notation
         (
             """
           Resources:
@@ -43,6 +43,30 @@ class TestCloudFormationLoader:
                   - PublicDnsName
           """,
             {"Resources": {"MyInstance": {"Type": "AWS::EC2::Instance", "Properties": {"UserData": GetAttTag(["MyInstance", "PublicDnsName"])}}}},
+        ),
+        # GetAtt tag tests - dot notation
+        (
+            """
+          Resources:
+            MyInstance:
+              Type: AWS::EC2::Instance
+              Properties:
+                UserData: !GetAtt MyInstance.PublicDnsName
+          """,
+            {"Resources": {"MyInstance": {"Type": "AWS::EC2::Instance", "Properties": {"UserData": GetAttTag(["MyInstance", "PublicDnsName"])}}}},
+        ),
+        # GetAtt tag tests - dot notation with nested attributes
+        (
+            """
+          Resources:
+            MyFunction:
+              Type: AWS::Lambda::Function
+              Properties:
+                Environment:
+                  Variables:
+                    ROLE_ARN: !GetAtt HelloWorldFunctionRole.Arn
+          """,
+            {"Resources": {"MyFunction": {"Type": "AWS::Lambda::Function", "Properties": {"Environment": {"Variables": {"ROLE_ARN": GetAttTag(["HelloWorldFunctionRole", "Arn"])}}}}}},
         ),
         # Sub tag tests
         (
@@ -171,7 +195,16 @@ class TestCloudFormationLoader:
           Properties:
             BucketName: !Ref
       """,
-        # Invalid GetAtt tag (wrong number of arguments)
+        # Invalid GetAtt tag (array notation with wrong number of arguments)
+        """
+      Resources:
+        MyInstance:
+          Type: AWS::EC2::Instance
+          Properties:
+            UserData: !GetAtt 
+              - MyInstance
+      """,
+        # Invalid GetAtt tag (dot notation without dot)
         """
       Resources:
         MyInstance:
@@ -431,6 +464,30 @@ class TestCloudFormationTemplateProcessor:
                 BUCKET_ARN: !GetAtt
                   - MyBucket
                   - Arn
+      """
+        template = load_yaml(yaml_content)
+
+        processor = CloudFormationTemplateProcessor(template)
+        processor.remove_resource("MyFunction")
+
+        # MyBucket should be removed
+        assert "MyBucket" not in processor.processed_template["Resources"]
+        assert "MyFunction" not in processor.processed_template["Resources"]
+
+    def test_remove_dependencies_get_att_dot_notation(self):
+        """Test removing dependencies with !GetAtt using dot notation."""
+        yaml_content = """
+      Resources:
+        MyBucket:
+          Type: AWS::S3::Bucket
+          Properties:
+            BucketName: my-bucket
+        MyFunction:
+          Type: AWS::Lambda::Function
+          Properties:
+            Environment:
+              Variables:
+                BUCKET_ARN: !GetAtt MyBucket.Arn
       """
         template = load_yaml(yaml_content)
 

